recipes/sagemaker-code-editor/build.sh

```diff
@@ -2,15 +2,64 @@
 
 set -exuo pipefail
 
+# This code includes code from https://github.com/conda-forge/openvscode-server-feedstock
+# which is licensed under the BSD-3-Clause License.
+
+rm $PREFIX/bin/node
+ln -s $BUILD_PREFIX/bin/node $PREFIX/bin/node
+
+export NODE_OPTIONS=--openssl-legacy-provider
+
+pushd sagemaker-code-editor
+pushd src
+
+# Install remote extensions for target_platform
+pushd remote
+  VSCODE_RIPGREP_VERSION=$(jq -r '.dependencies."@vscode/ripgrep"' package.json)
+  # Install all dependencies except @vscode/ripgrep
+  mv package.json package.json.orig
+  jq 'del(.dependencies."@vscode/ripgrep")' package.json.orig > package.json
+
+  yarn install
+  # Install @vscode/ripgrep without downloading the pre-built ripgrep.
+  # This often runs into Github API ratelimits and we won't use the binary in this package anyways.
+  yarn add --ignore-scripts "@vscode/ripgrep@${VSCODE_RIPGREP_VERSION}"
+popd
+
+# Install build tools for build_platform
+(
+  unset CFLAGS
+  unset CXXFLAGS
+  unset CPPFLAGS
+  unset npm_config_arch
+  export CC=${CC_FOR_BUILD}
+  export CXX=${CXX_FOR_BUILD}
+  export AR="$($CC_FOR_BUILD -print-prog-name=ar)"
+  export NM="$($CC_FOR_BUILD -print-prog-name=nm)"
+  export LDFLAGS=${LDFLAGS//$PREFIX/$BUILD_PREFIX}
+  export PKG_CONFIG_PATH=${BUILD_PREFIX}/lib/pkgconfig
+  VSCODE_RIPGREP_VERSION=$(jq -r '.dependencies."@vscode/ripgrep"' package.json)
+  VSCODE_TELEMETRY_VERSION=$(jq -r '.devDependencies."@vscode/telemetry-extractor"' package.json)
+  # Install all dependencies except @vscode/ripgrep
+  mv package.json package.json.orig
+  jq 'del(.dependencies."@vscode/ripgrep")' package.json.orig | jq 'del(.devDependencies."@vscode/telemetry-extractor")' > package.json
+  yarn install
+  # Install @vscode/ripgrep without downloading the pre-built ripgrep.
+  # This often runs into Github API ratelimits and we won't use the binary in this package anyways.
+  yarn add --ignore-scripts "@vscode/ripgrep@${VSCODE_RIPGREP_VERSION}" "@vscode/telemetry-extractor@${VSCODE_TELEMETRY_VERSION}"
+)
+ARCH_ALIAS=linux-x64
+yarn gulp vscode-reh-web-${ARCH_ALIAS}-min
+popd
+
 mkdir -p $PREFIX/share
-cp -R sagemaker-code-editor $PREFIX/share/
+cp -r vscode-reh-web-${ARCH_ALIAS} ${PREFIX}/share/sagemaker-code-editor
+rm -rf $PREFIX/share/sagemaker-code-editor/bin
```

Comment:
I am not familiar with this way of building things, but this sequence seems surprising:
 - build in place
 - copy resulting directory to share
 - move the binary parts to the PREFIX/in directory
 
 Rather than a regular:
  - configure and build in a separate directory
  - run the install command of the build tool to parsimously installing the bits that are required.

Reply:
To our knowledge, the Code OSS code base does not have an install command for the code server web option. Thus, we need to build the source code and create an executable that invokes the relevant script to start the server. 
Please let me know if you'd have any other suggestion here.